### PR TITLE
DeploymentConfig Unit Tests

### PR DESCRIPTION
**Pull request by ananth102**
Issue #, if available:

Description of changes:
Contains unit test for ReadOne operation of Deployment Config.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by surajkota — CHANGES_REQUESTED** [review]
+1, can you add one for create as well.

Thanks

**Comment by ananth102** [comment]
\retest

**Comment by ananth102** [comment]
retest

File: pkg/resource/endpoint/testdata/v1alpha1/create/observed/success_after_create_deployment_config.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    blueGreenUpdatePolicy:
+      maximumExecutionTimeoutInSeconds: 600
+      terminationWaitInSeconds: 100
+      trafficRoutingConfiguration:
+        linearStepSize:
+          type_: INSTANCE_COUNT
+          value: 1
+        type_: LINEAR
+        waitIntervalInSeconds: 100
+status:
```
Comment (surajkota) [inline pkg/resource/endpoint/testdata/v1alpha1/create/observed/success_after_create_deployment_config.yaml:22]: missing several fields

Comment (ananth102) [reply pkg/resource/endpoint/testdata/v1alpha1/create/observed/success_after_create_deployment_config.yaml:22]: This is right after the create call and before the subsequent readOne call, the controller wont know the status of the endpoint.

**Pull request merged**